lib/internal/util/inspect.js

```diff
@@ -1320,12 +1323,50 @@ function markCwd(ctx, line, workingDirectory) {
   return tempLine;
 }
 
-function safeGetCWD() {
+// The following characters are percent-encoded when converting from file path
+// to URL:
+// - %: The percent character is the only character not encoded by the
+//        `pathname` setter.
+// - \: Backslash is encoded on non-windows platforms since it's a valid
+//      character but the `pathname` setters replaces it by a forward slash.
+// - LF: The newline character is stripped out by the `pathname` setter.
+//       (See whatwg/url#419)
+// - CR: The carriage return character is also stripped out by the `pathname`
+//       setter.
+// - TAB: The tab character is also stripped out by the `pathname` setter.
+const percentRegEx = /%/g;
+const backslashRegEx = /\\/g;
+const newlineRegEx = /\n/g;
+const carriageReturnRegEx = /\r/g;
+const tabRegEx = /\t/g;
+
+function encodePathChars(filepath) {
+  if (StringPrototypeIncludes(filepath, '%'))
```

Comment:
this seems like it could be done with a single global regex replace, and a function replacer?

Reply:
I'll check how that performs.

Reply:
I just checked and using replace directly is slower. Using test with a regex is faster in case there's nothing to replace and it's slower in case there is something to replace (at least in my mini benchmark). A path is relatively short and as such, a hand written replacer is very likely the fasted. I guess it's not that critical though and I would keep it as is for now.